airbyte-integrations/connectors/source-gitlab/source_gitlab/source.py

```diff
@@ -140,7 +141,35 @@ def _get_group_list(self, config: MutableMapping[str, Any]) -> List[str]:
     def _is_http_allowed() -> bool:
         return os.environ.get("DEPLOYMENT_MODE", "").upper() != "CLOUD"
 
-    def check_connection(self, logger, config) -> Tuple[bool, any]:
+    def _try_refresh_access_token(self, logger, config: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        This method attempts to refresh the expired `access_token`, while `refersh_token` is stil valid.
+        In order to obtain the new `refresh_token`, the Customer should `re-auth` in the source settings.
+        """
+        # get currernt authenticator
+        authenticator: Union[SingleUseRefreshTokenOauth2Authenticator, TokenAuthenticator] = self.__auth_params.get("authenticator")
+        try:
+            creds = authenticator.refresh_access_token()
+            # update the actual config values
+            config["credentials"]["access_token"] = creds[0]
+            config["credentials"]["refresh_token"] = creds[3]
+            config["credentials"]["token_expiry_date"] = authenticator.get_new_token_expiry_date(creds[1], creds[2]).to_rfc3339_string()
+            # update the config
+            emit_configuration_as_airbyte_control_message(config)
```

Comment:
isn't it the same what's implemented in the [Airbyte CDK](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/oauth.py#L224)?

Reply:
The  CDK version of this now doesn't attempt to refresh revoked or invalid `access_token` it doesn't refresh the `access_token` until the time comes for it to be expired. We explicitly trigger the `refresh`  procedure,

Reply:
Thanks for the clarification. Can access tokens get revoked without revoking refresh tokens?
l think [this](https://github.com/airbytehq/airbyte/pull/31207) update should be enough, however other reviewers might have different thoughts